Use golang image for postsubmit job

dhiller opened the pull request:
A [nmo postsubmit failed due to missing go](https://prow.ci.kubevirt.io/job-history/gs/kubevirt-prow/logs/push-tagged-node-maintenance-operator). This PR changes the job definition to use the golang image (which is based on our bootstrap image)

/cc @slintes

dhiller commented:
Currently running the job with golang image locally using `phaino`. Looking good so far.

dhiller commented:
Hm, not sure why but it yielded a 503:

```
go get: added sigs.k8s.io/kustomize/kustomize/v3 v3.8.7
./bin/operator-sdk generate kustomize manifests -q
./bin/operator-sdk: line 1: syntax error near unexpected token `<'
./bin/operator-sdk: line 1: `<html><body><h1>503 Service Unavailable</h1>'
make: *** [Makefile:213: bundle] Error 2
make: *** [Makefile:289: container-build] Error 2
+ EXIT_VALUE=2
```

slintes commented:
> Hm, not sure why but it yielded a 503:
> 
> ```
> go get: added sigs.k8s.io/kustomize/kustomize/v3 v3.8.7
> ./bin/operator-sdk generate kustomize manifests -q
> ./bin/operator-sdk: line 1: syntax error near unexpected token `<'
> ./bin/operator-sdk: line 1: `<html><body><h1>503 Service Unavailable</h1>'
> make: *** [Makefile:213: bundle] Error 2
> make: *** [Makefile:289: container-build] Error 2
> + EXIT_VALUE=2
> ```

Hey Daniel, thanks a lot for looking into this.
Is this error permanent? 🤔 

I'm also wondering why the job needs go at all... will check

dhiller commented:
> > Hm, not sure why but it yielded a 503:
> > ```
> > go get: added sigs.k8s.io/kustomize/kustomize/v3 v3.8.7
> > ./bin/operator-sdk generate kustomize manifests -q
> > ./bin/operator-sdk: line 1: syntax error near unexpected token `<'
> > ./bin/operator-sdk: line 1: `<html><body><h1>503 Service Unavailable</h1>'
> > make: *** [Makefile:213: bundle] Error 2
> > make: *** [Makefile:289: container-build] Error 2
> > + EXIT_VALUE=2
> > ```
> 
> Hey Daniel, thanks a lot for looking into this. Is this error permanent? thinking
> 
> I'm also wondering why the job needs go at all... will check

Hey @slintes, sorry for posting the wrong link, I've updated the PR description. To answer your question: yes, that failures seems to be occurring permanently.

Ping me if I can help here.

dhiller commented:
So I've run the job again using `phaino`, and this was the result:
```
...
make docker-push IMG=quay.io/kubevirt/node-maintenance-operator-bundle:v0.11.3
make[1]: Entering directory '/home/prow/go/src/github.com/kubevirt/node-maintenance-operator'
docker push quay.io/kubevirt/node-maintenance-operator-bundle:v0.11.3
The push refers to repository [quay.io/kubevirt/node-maintenance-operator-bundle]
ac465974b9f9: Preparing
e73e6a07bf15: Preparing
db1ed9ec74a2: Preparing
e73e6a07bf15: Pushed
db1ed9ec74a2: Pushed
ac465974b9f9: Pushed
v0.11.3: digest: sha256:9c79bfe321967a7a6e2d341de03d8ce3f2f3c488cefd8f67a04b99df831df13f size: 940
make[1]: Leaving directory '/home/prow/go/src/github.com/kubevirt/node-maintenance-operator'
./bin/opm index add --container-tool docker --mode semver --tag quay.io/kubevirt/node-maintenance-catalog:v0.11.3 --bundles quay.io/kubevirt/node-maintenance-operator-bundle:v0.11.3 
time="2022-03-02T11:50:22Z" level=info msg="building the index" bundles="[quay.io/kubevirt/node-maintenance-operator-bundle:v0.11.3]"
time="2022-03-02T11:50:22Z" level=info msg="running /bin/docker pull quay.io/kubevirt/node-maintenance-operator-bundle:v0.11.3" bundles="[quay.io/kubevirt/node-maintenance-operator-bundle:v0.11.3]"
time="2022-03-02T11:50:24Z" level=info msg="running docker create" bundles="[quay.io/kubevirt/node-maintenance-operator-bundle:v0.11.3]"
time="2022-03-02T11:50:24Z" level=info msg="running docker cp" bundles="[quay.io/kubevirt/node-maintenance-operator-bundle:v0.11.3]"
time="2022-03-02T11:50:24Z" level=info msg="running docker rm" bundles="[quay.io/kubevirt/node-maintenance-operator-bundle:v0.11.3]"
time="2022-03-02T11:50:24Z" level=info msg="Could not find optional dependencies file" dir=bundle_tmp296961814 file=bundle_tmp296961814/metadata load=annotations
time="2022-03-02T11:50:24Z" level=info msg="found csv, loading bundle" dir=bundle_tmp296961814 file=bundle_tmp296961814/manifests load=bundle
time="2022-03-02T11:50:24Z" level=info msg="loading bundle file" dir=bundle_tmp296961814/manifests file=node-maintenance-operator-controller-manager-metrics-service_v1_service.yaml load=bundle
time="2022-03-02T11:50:24Z" level=info msg="loading bundle file" dir=bundle_tmp296961814/manifests file=node-maintenance-operator-metrics-reader_rbac.authorization.k8s.io_v1_clusterrole.yaml load=bundle
time="2022-03-02T11:50:24Z" level=info msg="loading bundle file" dir=bundle_tmp296961814/manifests file=node-maintenance-operator-webhook-service_v1_service.yaml load=bundle
time="2022-03-02T11:50:24Z" level=info msg="loading bundle file" dir=bundle_tmp296961814/manifests file=node-maintenance-operator.clusterserviceversion.yaml load=bundle
time="2022-03-02T11:50:24Z" level=info msg="loading bundle file" dir=bundle_tmp296961814/manifests file=nodemaintenance.kubevirt.io_nodemaintenances.yaml load=bundle
time="2022-03-02T11:50:24Z" level=info msg="Generating dockerfile" bundles="[quay.io/kubevirt/node-maintenance-operator-bundle:v0.11.3]"
time="2022-03-02T11:50:24Z" level=info msg="writing dockerfile: index.Dockerfile239045956" bundles="[quay.io/kubevirt/node-maintenance-operator-bundle:v0.11.3]"
time="2022-03-02T11:50:24Z" level=info msg="running docker build" bundles="[quay.io/kubevirt/node-maintenance-operator-bundle:v0.11.3]"
time="2022-03-02T11:50:24Z" level=info msg="[docker build -f index.Dockerfile239045956 -t quay.io/kubevirt/node-maintenance-catalog:v0.11.3 .]" bundles="[quay.io/kubevirt/node-maintenance-operator-bundle:v0.11.3]"
make docker-push IMG=quay.io/kubevirt/node-maintenance-catalog:v0.11.3
make[1]: Entering directory '/home/prow/go/src/github.com/kubevirt/node-maintenance-operator'
docker push quay.io/kubevirt/node-maintenance-catalog:v0.11.3
The push refers to repository [quay.io/kubevirt/node-maintenance-catalog]
2b1d1bed375d: Preparing
16a6b5b421af: Preparing
26895fdeb37e: Preparing
a98a386b6ec2: Preparing
4e7f383eb531: Preparing
bc276c40b172: Preparing
bc276c40b172: Waiting
26895fdeb37e: Layer already exists
16a6b5b421af: Layer already exists
4e7f383eb531: Layer already exists
a98a386b6ec2: Layer already exists
bc276c40b172: Layer already exists
2b1d1bed375d: Pushed
v0.11.3: digest: sha256:2b1c457b7ea89a74114f1d2a1219634451532b19302c997c2e2be98992a88f7f size: 1578
make[1]: Leaving directory '/home/prow/go/src/github.com/kubevirt/node-maintenance-operator'
/home/prow/go/src/github.com/kubevirt/node-maintenance-operator/bin/controller-gen "crd:trivialVersions=true,preserveUnknownFields=false" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
vendor/k8s.io/client-go/plugin/pkg/client/auth/exec/metrics.go:21:2: cannot find package "." in:
        /home/prow/go/src/github.com/kubevirt/node-maintenance-operator/vendor/io/fs
Error: not all generators ran successfully
run `controller-gen crd:trivialVersions=true,preserveUnknownFields=false rbac:roleName=manager-role webhook paths=./... output:crd:artifacts:config=config/crd/bases -w` to see all available markers, or `controller-gen crd:trivialVersions=true,preserveUnknownFields=false rbac:roleName=manager-role webhook paths=./... output:crd:artifacts:config=config/crd/bases -h` for usage
make: *** [Makefile:121: manifests] Error 1
+ EXIT_VALUE=2
+ set +o xtrace
Cleaning up after docker in docker.
================================================================================
Cleaning up after docker
/usr/local/bin/runner.sh: line 50: wait: pid 9 is not a child of this shell
================================================================================
Done cleaning up after docker in docker.
ERRO[0572] FAIL                                          duration=9m32.756668268s error="exit status 2" job=push-tagged-node-maintenance-operator
WARN[0572] Aborting other jobs...                        duration=9m32.756668268s job=push-tagged-node-maintenance-operator
FATA[0572] FAILED                                        error="cancel: context canceled"
```

So no, the 503 seems to have been an intermittent thing.

slintes commented:
`cannot find package "." in:
        /home/prow/go/src/github.com/kubevirt/node-maintenance-operator/vendor/io/fs`

This error looks familiar, I think it's related to a wrong Go version (1.16 vs. 1.17).
Let me see how to get rid of Go usage at all.
Thx
